src/components/structures/MessagePanel.tsx

```diff
@@ -443,6 +448,11 @@ export default class MessagePanel extends React.Component<IProps, IState> {
         // Always show highlighted event
         if (this.props.highlightedEventId === mxEv.getId()) return true;
 
+        const threadingEnabled = SettingsStore.getValue("experimentalThreadSupport");
+        if (threadingEnabled && mxEv.replyEventId && this.props.hideThreadedMessages === true) {
```

Comment:
if possible, I'd push the `SettingsStore.getValue()` check as late as possible into the conditional for performance reasons.